### ISSUE TITLE
Equivalent of tf.range or theano.tensor.arange in CNTK?

### ISSUE THREAD

**Issue by kestrelm** [Microsoft/CNTK#2391]
Hello,

I couldn't find the equivalent of  tf.range or theano.tensor.arange in CNTK. Is there some way to do this in the CNTK Python framework? If not, can we add this feature for future releases?

Thanks

**Comment by ebarsoumMS** [comment]
Hi @kestrelm  can you provide more details on the scenario? Why can't you do it in Python and pass the result to C.constant?

**Comment by kestrelm** [comment]
Hello,
Say I have the following loss function in Theano:

```
     import keras.backend as K
        a0 = K.cast(y_true[:,0], 'int32')
        q0 = y_true[:,1]        
       qp = y_pred[K.arange(K.shape(y_pred)[0]), a0]

       err = q0 - K.reshape(qp, q0.shape)
        return K.mean(err*err, axis=-1)

```
In it, I need to access:  y_pred[K.arange(K.shape(y_pred)[0]), a0] .
So this means I want to have be able to access a partial slice of y_pred from a generated group of indices based on the shape of the tensor. It seems like this isn't possible in the current CNTK python bindings?

Thanks

**Comment by christopher5106** [comment]
For position embedding, I need to have a dynamic arange operator. Thank you
